runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -0,0 +1,1529 @@
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	const J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+	jint result = 0;
+	J9UTF8 *name = NULL;
+	J9UTF8 *sig = NULL;
+	j9object_t callerObject = ((NULL == caller) ? NULL : J9_JNI_UNWRAP_REFERENCE(caller));
+
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	Trc_JCL_java_lang_invoke_MethodHandleNatives_getMembers_Entry(env, defc, matchName, matchSig, matchFlags, caller, skip, results);
+
+	if ((NULL == defc) || (NULL == results) || ((NULL != callerObject) && (J9VMJAVALANGCLASS(vm) != J9OBJECT_CLAZZ(currentThread, callerObject)))) {
+		result = -1;
+	} else {
+		j9object_t defcObject = J9_JNI_UNWRAP_REFERENCE(defc);
+		J9Class *defClass = J9VM_J9CLASS_FROM_HEAPCLASS(currentThread, defcObject);
+
+		if (NULL != matchName) {
+			name = vmFuncs->copyStringToJ9UTF8WithMemAlloc(currentThread, J9_JNI_UNWRAP_REFERENCE(matchName), J9_STR_NONE, "", 0, NULL, 0);
+			if (NULL == name) {
+				vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+				goto done;
+			}
+		}
+		if (NULL != matchSig) {
+			sig = vmFuncs->copyStringToJ9UTF8WithMemAlloc(currentThread, J9_JNI_UNWRAP_REFERENCE(matchSig), J9_STR_NONE, "", 0, NULL, 0);
+			if (NULL == sig) {
+				vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+				goto done;
+			}
+		}
+
+		if (!(((NULL != matchName) && (0 == J9UTF8_LENGTH(name))) || ((NULL != matchSig) && (0 == J9UTF8_LENGTH(sig))))) {
+			j9array_t resultsArray = (j9array_t)J9_JNI_UNWRAP_REFERENCE(results);
+			UDATA length = J9INDEXABLEOBJECT_SIZE(currentThread, resultsArray);
+			UDATA index = 0;
+
+			if (J9ROMCLASS_IS_INTERFACE(defClass->romClass)) {
+				result = -1;
+			} else {
+				if (J9_ARE_ANY_BITS_SET(matchFlags, MN_IS_FIELD)) {
+					J9ROMFieldShape *romField = NULL;
+					J9ROMFieldWalkState walkState;
+
+					UDATA classDepth = 0;
+					if (J9_ARE_ANY_BITS_SET(matchFlags, MN_SEARCH_SUPERCLASSES)) {
+						/* walk superclasses */
+						J9CLASS_DEPTH(defClass);
+					}
+					J9Class *currentClass = defClass;
+
+					while (NULL != currentClass) {
+						/* walk currentClass */
+						memset(&walkState, 0, sizeof(walkState));
+						romField = romFieldsStartDo(currentClass->romClass, &walkState);
+
+						while (NULL != romField) {
+							J9UTF8 *nameUTF = J9ROMFIELDSHAPE_NAME(romField);
+							J9UTF8 *signatureUTF = J9ROMFIELDSHAPE_SIGNATURE(romField);
+
+							if (((NULL == matchName) || J9UTF8_EQUALS(name, nameUTF))
+							&& ((NULL == matchSig) || J9UTF8_EQUALS(sig, signatureUTF))
+							) {
+								if (skip > 0) {
+									skip -=1;
+								} else {
+									if (index < length) {
+										/* Refetch reference after GC point */
+										resultsArray = (j9array_t)J9_JNI_UNWRAP_REFERENCE(results);
+										j9object_t memberName = J9JAVAARRAYOFOBJECT_LOAD(currentThread, resultsArray, index);
+										if (NULL == memberName) {
+											result = -99;
+											goto done;
```

Comment:
Why `-99`?  Is there a meaning to this value?

Reply:
This is the value used by OJDK impl for this particular error case